lib/tpm2_options.c

```diff
@@ -457,6 +458,12 @@ tpm2_option_code tpm2_handle_options (int argc, char **argv,
         case 'Z':
             flags->enable_errata = 1;
             break;
+        case 'X':
+            if (tool_opts) {
+                flags->no_tpm = 1;
+                tool_opts->flags |= TPM2_OPTIONS_NO_SAPI;
+            }
+            break;
```

Comment:
I missed this coming into master. This is broken and we need not to introduce this on 3.X. Please make this a makecredential and whatever other tools actually can make use of it specific change.

See: https://github.com/tpm2-software/tpm2-tools/issues/1442

Reply:
I think we can move the option from the main tools list into the makecredential option list: 
https://github.com/tpm2-software/tpm2-tools/blob/9ea358e464c3b2885b926b66c0493c5bd93f2611/lib/tpm2_options.c#L334-L342

But I'm not sure how we can move the logic into makecredential from options.c: 
https://github.com/tpm2-software/tpm2-tools/blob/9ea358e464c3b2885b926b66c0493c5bd93f2611/lib/tpm2_options.c#L408-L413

Since SAPI is set up a few lines later (which requires a TPM): 
https://github.com/tpm2-software/tpm2-tools/blob/9ea358e464c3b2885b926b66c0493c5bd93f2611/lib/tpm2_options.c#L448-L452

There doesn't appear to be any way to modify the `tool_opts->flags` variable from within the tools themselves except for in `tpm2_tool_onrun`, but it is too late by then since SAPI will already have been set up. 

If we want to move the logic into makecredential, the only thing I can think of is to allow `on_option` to take in a pointer to `tool_opts->flags` for all tools.  Do you know of another option, or should I try this and update all of the tpm2 tools `on_option` functions?

Reply:
You can modify the flags by setting them when setting your on-option handler via:
tpm2_options_new() @ https://github.com/tpm2-software/tpm2-tools/blob/3.X/lib/tpm2_options.h#L142

Reply:
Oh they need to be modified post option handler init. We can just make a routine into options to set that flag.

Reply:
I almost wonder, rather than making it an option like -X, we allow an empty `-T` or `-T-` to say hey no TCTI and set a flag that says TCTI can be optional. I like that better.

Reply:
Feel free to take the changes on:
https://github.com/tpm2-software/tpm2-tools/pull/1445

and backport them to 3.X. I'd wait for the review period to close to see if anyone spots any issues or loudly objects.

Reply:
I like that solution ... I'll play around with the patch on my system.

Reply:
@williamcroberts I did something similar to your patch for master.  Let me know what you think.